### PR TITLE
power utilites to order 2,3,4

### PR DESCRIPTION
**Pull request by kostyfisik**
The rationale for this PR was proposed in #3227

### CODE REVIEW

**Comment by tjhei** [comment]
> This functions have less code compared to
> <code>fixed_power@<dim@> (n)</code> 

That shouldn't matter.

> and should be a bit faster to
> compile (less optimizations needed) 

Well, you are now compiling more code when including utilities.h or not?

> and faster to run (in debug
> mode).

Did you test this and why does this matter to you?

I also find the naming of this function somewhat unintuitive and think ``fixed_power<2>`` is much better in that regard.

File: include/deal.II/base/utilities.h [hunk 1]
```diff
@@ -270,6 +270,26 @@ namespace Utilities
...
+   * Calculate a fixed power (second, third, and forth order)  of a number.
+   *
+   * This functions have less code compared to
+   * <code>fixed_power@<dim@> (n)</code> and should be a bit faster to
+   * compile (less optimizations needed) and faster to run (in debug
+   * mode).
+   */
+  template <typename T>
+  T
+  pow2 (const T t);
```
Comment (davydden) [inline include/deal.II/base/utilities.h]: my personal opinion is that `fixed_power<2>()` is as readable as `pow2`, but I let others to decide on this issue.

**Comment by kronbichler** [comment]
I also do not really see the advantage of this. Even though I find it annoying from time to time to write `Utilities::fixed_power<2>(a)` rather than `pow2(a)`, it would find it at least as annoying to learn and remember yet another function name - or find that other function name among the suggestions of Eclipse (especially when it is in the global namespace).

**Comment by davydden** [comment]
A fair comparison is actually `Utilities::fixed_power<2>(a)` vs `Utilities::pow2(a)`, unless you expose `Utilities` namespace of deal.II.

**Comment by kostyfisik** [comment]
Ok, as I already mentioned it is not a problem for me to copy-paste one-liner... As soon as profit is not obvious for 3 of 4 devs - it does not worth adding to dealii.

**Pull request closed without merging**